packages/runtime/src/tutorial-runner.ts

```diff
@@ -302,6 +302,43 @@ export class TutorialRunner {
     );
   }
 
+  /**
+   * Get snapshot of runner's current files.
+   * Also prepares `package.json`'s `stackblitz.startCommand` with runner's commands.
+   *
+   * Note that file paths do not contain the leading `/`.
+   */
+  takeSnapshot() {
+    const files: Record<string, string> = {};
+
+    // first add template files
+    for (const [filePath, value] of Object.entries(this._currentTemplate || {})) {
+      if (typeof value === 'string') {
+        files[removeLeadingSlash(filePath)] = value;
+      }
+    }
+
+    // next overwrite with files from editor
+    for (const [filePath, value] of Object.entries(this._currentFiles || {})) {
+      if (typeof value === 'string') {
+        files[removeLeadingSlash(filePath)] = value;
+      }
+    }
+
+    const packageJson = parseJson(files['package.json']);
+
+    // add start commands
+    if (files['package.json']) {
```

Comment:
This should check if there is already a start or dev script because then we don't need to do anything. StackBlitz default logic will be good enough I think.